cfy_manager/networks/networks.py

```diff
@@ -93,8 +114,10 @@ def add_networks(networks=None,
         create_internal_certs()
 
     script_path = join(SCRIPT_DIR, 'update-manager-networks.py')
+    hostname = metadata['hostname'] if metadata.get('hostname')\
```

Comment:
```suggestion
    hostname = metadata.get('hostname') or socket.gethostname()
```
This way you won't need to add the hostname to the config file. Unless I missed something, and it needs to be there?

Notice, this means you would also need to add `import socket` as part of the imports.

Reply:
@YoniItzhak if you check at the beginning of this method, I'm calling `_set_hostname()` which already populate the the `config[MANAGER][HOSTNAME]` so value will be there if the its not part of the config file

Reply:
```def _set_hostname():
    if not config[MANAGER][HOSTNAME]:
        config[MANAGER][HOSTNAME] = socket.gethostname()
```

Reply:
I'm stating that you don't need to change it in the config.yaml. If I understand correctly, all you want is the instance hostname. So just get it by using `socket.gethostname()`, you don't need to put it in the config.yaml too.

Reply:
well, it could be part of the config file and provided my user I cannot just use `socket.gethostname()`

Reply:
so basically configuration file could have this file so why do I need to use the `socket.gethostname()` ? set_hostname will handle that and get the value first from config file and if it does not exist it get it from hostname

Reply:
I see. This is one way to do it, but this function is supposed to add a network, not to add the hostname to the config.yaml file. This way, we get a strange behavior where you run `add_network` and then all of a sudden you also see the hostname in the config.yaml file. 

I think a better way would be doing:
```
    def get_hostname():
      return config[MANAGER].get(HOSTNAME) or socket.gethostname()
```

Then, the code would be: 
```suggestion
    hostname = metadata.get('hostname') or get_hostname()
```

Reply:
it will not update the configuration file by the way it will just make the configuration of hostname available so instead of adding new method we can just use the set_host_name which is something just update the dict not the file

Reply:
Right, but this config is used afterwards buy other methods. I just think that a function is supposed to do the one thing it's created for, not more than that. There is no reason to set the hostname in the config and then use it if you can get it directly.